design/application-abi.md

```diff
@@ -0,0 +1,37 @@
+WASI Application ABI
+====================
+
+In addition to the APIs defined by the various WASI [modules](modules.md) there
+are also certain expectations that the WASI runtime places on an application
+that wishes to be portable across WASI implementations.
+
+This document describes how a conforming WASI application is expected to behave
+in terms of lifecycle (startup, shutdown, etc) and any exports it is expected to
+include.
+
+Current Unstable ABI
+--------------------
+
+The current WASI unstable ABI specifies only two exports from a WASI
+application:
+
+- `_start` - the program entry point
```

Comment:
@lukewagner @guybedford in https://github.com/nodejs/node/pull/27850#issuecomment-499475321 we discussed the issue of running the main on import. However, this assumption seems to be wrong? Nodejs could decide to expose the `_start` function to the user.